### ISSUE TITLE
Is private key file protected?

### ISSUE THREAD

**Issue by abakadaba** [tpm2-software/tpm2-openssl#94]
This is maybe a trivial question but since I couldn't find any explicit statement and I need to be absolutely sure about it: Is the private key file `testkey.priv`, that is generated by the command
`openssl genpkey -provider tpm2 -algorithm RSA -pkeyopt bits:1024 -out testkey.priv`,
protected, i.e. encrypted and HMACed with its parent TPM key?

**Comment by gotthardp** [comment]
Yes, it is. The private key in the TSS2 PEM file is protected by the TPM and its parent key. It's not a plain RSA/EC key.

**Comment by gotthardp** [comment]
Good point. There is now (latest `master` branch). The `-text` parameter should print also the parent, e.g.:

```
$ openssl pkey -provider tpm2 -provider base -in testkey.priv -noout -text
Private-Key: (RSA 1024 bit, TPM 2.0)
Parent: 0x40000001
Modulus:
    00:b2:cb:e9:af:ed:45:9d:8d:46:23:29:cf:14:dd:
    cf:a1:f4:40:5b:c8:e0:1a:35:8b:00:e9:e8:7a:4f:
    a9:d0:64:1e:f0:80:a9:b8:a3:f4:cd:3b:40:f4:7d:
    2c:61:7d:4b:c0:47:92:39:47:7d:0f:49:04:04:34:
    55:95:a1:ed:fa:84:89:cb:60:b2:ee:0e:ee:0e:4c:
    2b:78:e1:84:4a:09:6f:eb:8b:c9:14:82:6b:7e:1a:
    77:a6:f5:37:35:a8:60:81:7c:db:68:c5:28:87:ba:
    c7:7a:8f:16:32:65:5e:7b:e4:fe:f4:b8:39:02:5e:
    a9:9d:2a:cd:56:45:1e:5a:37
Exponent: 65537 (0x10001)
Object Attributes:
  fixedTPM
  fixedParent
  sensitiveDataOrigin
  userWithAuth
  restricted
  sign / encrypt
Signature Scheme: PSS
  Hash: SHA1
```

**Comment by abakadaba** [comment]
Nice, thanks! 🙂

**Comment by gotthardp** [comment]
Closed by bf4c7d6f9cfdc26444950b45bc0d1f1fb99afd4c and c773cd66c56d7df71a9dea13887e0d3da22a2cc3.
